py3status/modules/net_rate.py

```diff
@@ -50,39 +65,36 @@ class Py3status:
     devfile = '/proc/net/dev'
     format = "{interface}: {total}"
     format_no_connection = ''
+    format_value = "{value:5.1f} {unit:>5s}"
     hide_if_zero = False
-    interfaces = ''
+    interfaces = []
     interfaces_blacklist = 'lo'
-    precision = 1
+    si_units = False
+    thresholds = [(0, "bad"), (1024, "degraded"), (1024*1024, "good")]
+    unit = "B/s"
+    # obsolete configuration parameters
+    precision = None
 
     def __init__(self, *args, **kwargs):
-        """
-        Format of total, up and down placeholders under self.format.
-        As default, substitutes self.left_align and self.precision as %s and %s
-        Placeholders:
-            value - value (float)
-            unit - unit (string)
-        """
         self.last_interface = None
         self.last_stat = self._get_stat()
         self.last_time = time()
 
-    def currentSpeed(self):
+    def post_config_hook(self):
         # parse some configuration parameters
         if not isinstance(self.interfaces, list):
             self.interfaces = self.interfaces.split(',')
         if not isinstance(self.interfaces_blacklist, list):
             self.interfaces_blacklist = self.interfaces_blacklist.split(',')
 
-        # == 6 characters (from MULTIPLIER_TOP + dot + self.precision)
-        if self.precision > 0:
-            self.left_align = len(str(MULTIPLIER_TOP)) + 1 + self.precision
-        else:
-            self.left_align = len(str(MULTIPLIER_TOP))
-        self.value_format = "{value:%s.%sf} {unit}" % (
-            self.left_align, self.precision
-        )
+        if self.precision is not None:
+            if self.precision > 0:
+                self.left_align = 3 + 1 + self.precision
+            else:
+                self.left_align = 3
+            self.format_value = "{value:%s.%sf} {unit}" % (self.left_align, self.precision)
```

Comment:
This should be equivalent to the default if `precision` is set to `1` (which used to be the default). Of course, I need to fix this to be `{unit:>5s}` instead of `{unit}`. If the user has set `format_value`, then they shouldn't set `precision`.